src/main/java/net/mcreator/ui/workspace/resources/WorkspacePanelStructures.java

```diff
@@ -18,102 +18,36 @@
-
-		JSelectableList<String> structureElementList = new JSelectableList<>(filterModel);
-		structureElementList.setOpaque(false);
-		structureElementList.setCellRenderer(new Render());
-		structureElementList.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
-
-		JScrollPane sp = new JScrollPane(structureElementList);
-		sp.setOpaque(false);
-		sp.setHorizontalScrollBarPolicy(ScrollPaneConstants.HORIZONTAL_SCROLLBAR_NEVER);
-		sp.getViewport().setOpaque(false);
-		sp.getVerticalScrollBar().setUnitIncrement(11);
-		sp.getVerticalScrollBar().setUI(new SlickDarkScrollBarUI((Color) UIManager.get("MCreatorLAF.DARK_ACCENT"),
-				(Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"), sp.getVerticalScrollBar()));
-		sp.getVerticalScrollBar().setPreferredSize(new Dimension(8, 0));
-
-		add("Center", sp);
-
-		TransparentToolBar bar = new TransparentToolBar();
-		bar.setBorder(BorderFactory.createEmptyBorder(3, 5, 3, 0));
-
-		JButton importnbt = L10N.button("action.workspace.resources.import_structure");
-		importnbt.setIcon(UIRES.get("16px.open.gif"));
-		importnbt.setContentAreaFilled(false);
-		importnbt.setOpaque(false);
-		ComponentUtils.deriveFont(importnbt, 12);
-		importnbt.setBorder(BorderFactory.createEmptyBorder(0, 8, 0, 8));
-		bar.add(importnbt);
-
-		JButton importmc = L10N.button("action.workspace.resources.import_structure_from_minecraft");
-		importmc.setIcon(UIRES.get("16px.open.gif"));
-		importmc.setContentAreaFilled(false);
-		importmc.setOpaque(false);
-		ComponentUtils.deriveFont(importmc, 12);
-		importmc.setBorder(BorderFactory.createEmptyBorder(0, 8, 0, 8));
-		bar.add(importmc);
+		super(workspacePanel, new ResourceFilterModel<>(workspacePanel, String::toString), new Render());
 
-		JButton del = L10N.button("workspace.sounds.delete_selected");
-		del.setIcon(UIRES.get("16px.delete.gif"));
-		del.setOpaque(false);
-		del.setContentAreaFilled(false);
-		del.setBorder(BorderFactory.createEmptyBorder(0, 8, 0, 8));
-		bar.add(del);
-
-		del.addActionListener(a -> deleteCurrentlySelected(structureElementList));
-
-		structureElementList.addKeyListener(new KeyAdapter() {
-			@Override public void keyReleased(KeyEvent e) {
-				if (e.getKeyCode() == KeyEvent.VK_DELETE) {
-					deleteCurrentlySelected(structureElementList);
-				}
-			}
-		});
-
-		importnbt.addActionListener(e -> workspacePanel.getMCreator().actionRegistry.importStructure.doAction());
-		importmc.addActionListener(
+		addToolBarButton("action.workspace.resources.import_structure", UIRES.get("16px.open.gif"),
+				e -> workspacePanel.getMCreator().actionRegistry.importStructure.doAction());
+		addToolBarButton("action.workspace.resources.import_structure_from_minecraft", UIRES.get("16px.open.gif"),
 				e -> workspacePanel.getMCreator().actionRegistry.importStructureFromMinecraft.doAction());
-
-		add("North", bar);
-
+		addToolBarButton("workspace.sounds.delete_selected", UIRES.get("16px.delete.gif"),

```

Comment:
This should be a separate key (given you refactor this class anyway; I have this fixed on usages search PRs too but this one feels more related).

Reply:
What do you mean?

Reply:
I think he meant to use the same common key for all cases of "Delete selected"

Reply:
Oh ok (this is what I thought was the best, but I thought he meant using multiple keys, so I didn't understand)